pyscript.core/src/stdlib/pyweb/pydom.py

```diff
@@ -34,6 +36,23 @@ def JsProxy(obj):
 alert = window.alert
 
 
+class JSProperty:
+    """JS property descriptor that directly maps to the property with the same
+    name in the underlying JS component."""
+
+    def __init__(self, name: str, allow_nones: bool = False):
+        self.name = name
+        self.allow_nones = allow_nones
+
+    def __get__(self, obj, objtype=None):
+        return getattr(obj._js, self.name)
+
+    def __set__(self, obj, value):
+        if not self.allow_nones and value is None:
+            return
```

Comment:
Suggestion (feel free to ignore): raise `ValueError` because `None` isn't a valid value if `self.allow_nones` is `False`..? Something like this..?

```python
if self.allow_nones and value is None:
    raise ValueError(f"Cannot set a None value for {self.name}.")
```

The current code appears to not have any side-effects, so the programmer will just assume their `None` value has been set, which is probably a dangerous assumption to make. 

> Errors should never pass silently.

(Zen of Python)